lib/Jsrt/Jsrt.cpp

```diff
@@ -1215,6 +1215,15 @@ CHAKRA_API JsStringToPointer(_In_ JsValueRef stringValue, _Outptr_result_buffer_
 
 CHAKRA_API JsConvertValueToString(_In_ JsValueRef value, _Out_ JsValueRef *result)
 {
+    if (value != nullptr && Js::JavascriptString::Is(value))
+    {
+        return ContextAPINoScriptWrapper_NoRecord([&](Js::ScriptContext *scriptContext) -> JsErrorCode {
+            *result = Js::CrossSite::MarshalVar(scriptContext, value,
```

Comment:
>result [](start = 13, length = 6)

result can be null, it will crash in that case?

Reply:
We should move the `PARAM_NOT_NULL(result);` from inside the `ContextAPIWrapper` to up front.

Reply:
We probably also want the `VALIDATE_INCOMING_REFERENCE(value, scriptContext)` from the next wrapper in here?

Reply:
> `result` can be null..

`PARAM_NOT_NULL(result)` check.. well, I'm okay adding all sort of those checks but what purpose this method has if the `result` argument is nullptr? Well, isn't that a programatical mistake?

Reply:
Yes it is, and the `PARAM_NOT_NULL` check is what reports that back to the embedder calling the function.

Reply:
Note that the check was already there originally, it was just inside the context wrapper.

Reply:
I know that. My point was, some stuff inside the JSRT.cpp feels like they are there because of copy-paste from another API method... This particular interface, what is difference between returning `hey result is null, so you called this interface with something you definitely shouldn't, probably you have something terribly wrong` vs `crash`) because, whatever the application logic is.. something is wrong.

Imagine..

scenario A)
```
JsValueRef result;
JsConvertValueToString(Some_JS_Value, &result)
```

if `&result is null?`... 

scenario b)
```
JsConvertValueToString(Some_JS_Value, nullptr)
```

why do you call this interface then?

scenario c)
```
JsConvertValueToString(Some_JS_Value, &someClass->result)
```

so... `someClass` is probably dead? It will crash anyway?

I will apply the checks..

Reply:
I would suggest that crashing with a reason is always more helpful than crashing without a reason. If there is something wrong with an embedder's logic, we should at least do simple checks like this to let them know that the crash is their fault, not ours.